---
title: "Exploring transcript data profile of healthy and AML patient samples"
output: 
    github_document:
        html_preview: false
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Loading the required libraries

```{r message=FALSE, warning=FALSE}
library(tximport)
library(tidyverse)
library(DESeq2)
library(RColorBrewer)
library(pheatmap)
```


## Loading the data

Prepare the sample table of the experiment and load quant data generated by Salmon into R

```{r prepare_data, message=FALSE, warning=FALSE}
sample_table <- read_csv("../data/SRP518774_metadata.txt") %>%
  filter(`Assay Type` == "RNA-Seq") %>%
  select(`Run`, `isolate`) %>%
  mutate(condition = if_else(str_detect(isolate, "health"), "Control/Healthy", "AML Patient")) %>%
  select(`Run`, condition) %>%
  mutate(sample_name = Run) %>%
  select(sample_name, condition)

sample_table

sample_files <- paste0(pull(sample_table, 
            sample_name), '_quant/quant.sf')

sample_files <- paste0("../data/quants/", sample_files)

#paste0(sample_files, '_quant/quant.sf')

gene_map <- read_csv('../data/gene_map.csv', 
                     col_names = c('enstid', 'ensgid'))

names(sample_files) <- pull(sample_table, sample_name)


count_data <- tximport(files = sample_files,
                       type = 'salmon', 
                       tx2gene = gene_map,
                       ignoreTxVersion = TRUE)

# peek out counts data (non-normalized) and trancript - gene map
count_data$counts %>% head(10)
gene_map %>% head(10)
```


## Load the quant data and sample information into DESeq2 datatype, and run the DESeq2 normalization

```{r deseq2, message=FALSE, warning=FALSE}
sample_table$condition <- factor((c('AML', 'AML', 'AML', 'AML', 'Healthy',
                                         'Healthy', 'AML', 'AML', 'Healthy', 'Healthy',
                                         'Healthy', 'Healthy', 'Healthy', 'AML', 'AML')),
                                      levels = c('AML', 'Healthy'))

dds_aml <- DESeqDataSetFromTximport(txi = count_data,
                                     colData = sample_table,
                                     design = ~condition)

# Normalization - median of ratios method TTM DESeq2
dds_aml <- estimateSizeFactors(dds_aml)
#normalizationFactors(dds_aml)
counts(dds_aml, normalized = T) %>% head(10)

```

## Principal Component Analysis

```{r pca, message=FALSE, warning=FALSE}
# Transform the data so it is suitable for PCA
vst_aml <- varianceStabilizingTransformation(dds_aml)
# create matrix
vst_mat <- assay(vst_aml)

pca <- prcomp(t(vst_mat))

# plot out the data frame to produce PCA
df <- as.data.frame(pca$x)
df$condition <- sample_table$condition

pve <- round(pca$sdev^2/sum(pca$sdev^2) * 100, 2)

rownames_to_column(df, var = "sample_name") %>% as.tibble() %>%
  ggplot(., aes(x=PC1, y=PC2, color = condition)) +
  geom_point(size = 4) +
  #geom_text(aes(label = sample_name, color = condition), vjust = -1, size = 4) +
  xlab(label = paste0("PC1 (", pve[1], "%)")) +
  ylab(label = paste0("PC2 (", pve[2], "%)")) +
  theme_classic() +
  ggtitle("Principal Component Analysis")

ggsave("../figures/PCA.png")

```

Healthy samples cluster closely together on the right side, forming a tight group distinct from the AML patient samples. AML samples appear to form two separate clusters along the PC2 axis (13.64% variance explained). Additionally, two overlapping healthy samples are represented by a single point so that we only see five dots for healthy samples.

## Heatmap

```{r message=FALSE, warning=FALSE}
distance <- dist(t(assay(vst_aml)))
distance_matrix <- as.matrix(distance)
rownames(distance_matrix) <- vst_aml$condition
colnames(distance_matrix) <- vst_aml$condition
colors <- colorRampPalette( rev(brewer.pal(9, "Reds")) )(255)

pheatmap(distance_matrix,
         clustering_distance_rows=distance,
         clustering_distance_cols=distance,
         col=colors)

ggsave("../figures/heatmap.png")


distance <- dist(t(assay(vst_aml)))
distance_matrix <- as.matrix(distance)
rownames(distance_matrix) <- vst_aml$condition
colnames(distance_matrix) <- vst_aml$sample_name
colors <- colorRampPalette( rev(brewer.pal(9, "Reds")) )(255)

pheatmap(distance_matrix,
         clustering_distance_rows=distance,
         clustering_distance_cols=distance,
         col=colors)

```


comnfirmed with the heatmap, two dots consist of two samples that very very similiar. Is that come from same person but different processed samples (technical replicates)? 

